File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/factory/BotHandshakeFactory.java
package dev.robocode.tankroyale.botapi.factory;

import dev.robocode.tankroyale.botapi.BotInfo;
import dev.robocode.tankroyale.schema.BotHandshake;

import java.util.ArrayList;

/** Utility class used fro creating bot handshakes. */
public final class BotHandshakeFactory {

  public static BotHandshake create(BotInfo botInfo) {
    BotHandshake handshake = new BotHandshake();
    handshake.set$type(BotHandshake.$type.BOT_HANDSHAKE);
    handshake.setName(botInfo.getName());
    handshake.setVersion(botInfo.getVersion());
    handshake.setAuthor(botInfo.getAuthor());
    handshake.setCountryCode(botInfo.getCountryCode());
    handshake.setGameTypes(new ArrayList<>(botInfo.getGameTypes()));
    handshake.setProgrammingLang(botInfo.getProgrammingLang());
    return handshake;
  }
}

File: robocode-tankroyale-bot-api-csharp/factory/BotHandshakeFactory.cs
using System.Collections.Generic;
using Robocode.TankRoyale.Schema;

namespace Robocode.TankRoyale.BotApi
{
  public class BotHandshakeFactory
  {
    public static BotHandshake Create(BotInfo botInfo)
    {
      var handshake = new BotHandshake();
      handshake.Type = MessageType.BotHandshake;
      handshake.Name = botInfo.Name;
      handshake.Version = botInfo.Version;
      handshake.Author = botInfo.Author;
      handshake.CountryCode = (botInfo.CountryCode);
      handshake.GameTypes = new List<string>(botInfo.GameTypes);
      handshake.ProgrammingLang = botInfo.ProgrammingLang;
      return handshake;
    }
  }
}